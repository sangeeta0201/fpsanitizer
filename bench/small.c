#include <stdio.h>
#include <math.h>

void calcY(double* y, double x){
  *y = sqrt(x + 1) - sqrt(x);
}

int main() {
  double x,y;
  x = 1e10;
  calcY(&y, x);
  printf("y:%e\n", y);
  calcY(&y, x);
  printf("y:%e\n", y);
  return 0;
}
