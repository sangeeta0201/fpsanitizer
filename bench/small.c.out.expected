(output
  (argIdx 0)
  (function "main")
  (filename "small.c")
  (line-num 13)
  (instr-addr 40067D)
  (avg-error 32.617943)
  (max-error 32.617943)
  (num-calls 1)
  (influences
    (
    (
     (expr
       (FPCore ()
          (- (sqrt (+ 1.000000 1.000000e10)) (sqrt 1.000000e10))))
     (var-problematic-ranges)
     (example problematic input ())
     (function "calcY")
     (filename "small.c")
     (line-num 5)
     (instr-addr 4005F9)
     (avg-error 32.617943)
     (max-error 32.617943)
     (avg-local-error 32.617943)
     (max-local-error 32.617943)
     (num-calls 2))
    )
  )
)
